File: Problem_3/problem_3.fs
open System

let rec problem_3 num  f =
    match num with
        | x when x <= 1L -> f
        | x when x % f = 0L ->  problem_3 (num / f) f
        | _ -> problem_3 num (f + 1L)

[<EntryPoint>]
let main(args : string[]) =
    printfn "result = %d" (problem_3 600851475143L 2L)
    0
    

File: Problem_03/problem_3.js
function problem_3(num) {
    var f = 2;

    while (num > 1 ) {
	if (num % f == 0)
	    num = num / f;
	else
	    f = f + 1;
    }

    return f;
}

document.write('<b>result: </b>' + problem_3(600851475143));
